### PR TITLE
Fix unresponsive textarea from addon-event

### LINKED ISSUE #6936: Addon Events: can't edit payload
https://storybooks-official.netlify.com/?path=/story/addons-events--logger

The field for editing event payload is unresponsive

### PR DESCRIPTION
**Pull request by lonyele**
Issue: Fixes #6936

#### Problem
  From `getDerivedStateFromProps`, comparing some props and some state(always `undefined`) always initializes state which makes changing on textarea unresponsive.

#### Implementation
- Assign payload value to prevPayload
- Using lodash `isEqual` to properly compare changing props' payload and state's prevPayload

## How to test

- Is this testable with Jest or Chromatic screenshots? 
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: addons/events/src/components/Event.js [hunk 1]
```diff
@@ -133,14 +134,14 @@ class Item extends Component {
...
       const payloadString = json.plain(payload);
-
+      const refinedPayload = getJSONFromString(payloadString);
       return {
         failed: false,
-        payload: getJSONFromString(payloadString),
+        payload: refinedPayload,
         payloadString,
-        prevPayload,
+        prevPayload: refinedPayload,
```
Comment (shilman) [inline addons/events/src/components/Event.js]: Why are payload & prevPayload the same thing?

**Review by shilman — CHANGES_REQUESTED** [review]
see comment

File: addons/events/src/components/Event.js (hunk 1 above)
Comment (lonyele) [reply addons/events/src/components/Event.js]: The reason for why `payload` and `prevPayload` exist is to determine whether the payload is changed or not(which is a prop and coming from user input from story)

This assigning value to `prevPayload` is only executed when coming `payload` and previous `payload`(which is the same as `prevPayload`)

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
@lonyele I invited you to the github org. 👍

File: addons/events/src/components/Event.js (hunk 1 above)
Comment (lonyele) [reply addons/events/src/components/Event.js]: @shilman I'm wondering if my explanation is not clear? just double-check

**Comment by lonyele** [comment]
@shilman just ping for the comment. I'd like to migrate addon-event to typescript...

File: addons/events/src/components/Event.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { polyfill } from 'react-lifecycles-compat';
+import { isEqual } from 'lodash';
```
Comment (ndelangen) [inline addons/events/src/components/Event.js]: This should be:
```js
import isEqual from 'lodash/isEqual';
```

Comment (ndelangen) [reply addons/events/src/components/Event.js]: @lonyele if you can resolve this, and the CI is green, you can merge!

**Comment by lonyele** [comment]
@ndelangen hm.. same goes here. Color is changed but it is not expected. I'll investigate this too

**Pull request merged**